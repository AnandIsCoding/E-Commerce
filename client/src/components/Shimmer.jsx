import React from 'react'

function Shimmer() {
  return (
    <div className='products flex-shrink-0 w-[45%] md:w-[18%] pb-0 p-2 md:px-2 md:pt-4 rounded-md relative overflow-y-scroll md:h-[60vh] '>
  {/* Image Shimmer */}
  <div className='w-full h-[50%] rounded-xl bg-gray-300 shimmer'></div>

  {/* Text Shimmers */}
  <div className='mt-2'>
    <p className='w-1/2 h-4 bg-gray-300 shimmer mx-auto rounded-md'></p>
    <p className='w-3/4 h-4 bg-gray-300 shimmer mt-2 rounded-md'></p>
    <p className='w-1/3 h-4 bg-gray-300 shimmer mx-auto mt-2 rounded-md'></p>
  </div>

  {/* Buttons Shimmers */}
  <div className='flex flex-col gap-2 mt-4'>
    <div className='w-full h-10 bg-gray-300 shimmer rounded-lg'></div>
    <div className='w-full h-10 bg-gray-300 shimmer rounded-lg mb-4'></div>
  </div>
</div>

  )
}

export default Shimmer
