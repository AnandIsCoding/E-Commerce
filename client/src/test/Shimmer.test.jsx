import { render, screen } from '@testing-library/react';
import Shimmer from '../components/Shimmer'
import { describe, it, expect } from 'vitest';
import React from 'react';

describe('Shimmer Component', () => {
  it('renders without crashing', () => {
    render(<Shimmer />);
    const shimmerDiv = screen.getByTestId('shimmercard'); // Check for any rendered div
    expect(shimmerDiv).toBeInTheDocument();
  });

  
});
