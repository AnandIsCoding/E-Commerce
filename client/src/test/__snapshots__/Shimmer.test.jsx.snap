// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shimmer Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="products flex-shrink-0 w-[45%] md:w-[18%] pb-0 p-2 md:px-2 md:pt-4 rounded-md relative overflow-y-scroll md:h-[60vh] "
  >
    <div
      class="w-full h-[50%] rounded-xl bg-gray-300 shimmer"
    />
    <div
      class="mt-2"
    >
      <p
        class="w-1/2 h-4 bg-gray-300 shimmer mx-auto rounded-md"
      />
      <p
        class="w-3/4 h-4 bg-gray-300 shimmer mt-2 rounded-md"
      />
      <p
        class="w-1/3 h-4 bg-gray-300 shimmer mx-auto mt-2 rounded-md"
      />
    </div>
    <div
      class="flex flex-col gap-2 mt-4"
    >
      <div
        class="w-full h-10 bg-gray-300 shimmer rounded-lg"
      />
      <div
        class="w-full h-10 bg-gray-300 shimmer rounded-lg mb-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
